### PR TITLE
[GSOC-46] Fuel model usage and migration to ROS 2 launch files

### PR DESCRIPTION
**Pull request by rakeshv24**
This PR introduces two features:

- **Fuel Model Integration Examples:** We've added comprehensive examples demonstrating how to utilize models from Fuel, making it easier to incorporate these models into your projects.

- **ROS 2 Launch File Migration:** Launch files have been updated to ROS 2, ensuring compatibility with the latest ROS version.

### CODE REVIEW

File: dave_demos/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+This method simplifies the process by pulling the model directly from Fuel, ensuring you always have the latest version without needing to manage local files.
+
+### 2. Launching a Dave Model using Downloaded Model Files
+
+If you prefer to use model files downloaded from Fuel, proceed as follows:
+
+1. Set the resource path to your local model files:
+   ```bash
+   export GZ_SIM_RESOURCE_PATH=<Path-to-dave_ws>/src/dave/dave_model_description
```
Comment (woensug-choi) [inline dave_demos/README.md]: https://github.com/gazebosim/ros_gz/pull/492 Could you try this to get rid of this step?

Comment (woensug-choi) [reply dave_demos/README.md]: I've no experience with `hooks` but it seems it's another alternative
Reference: https://github.com/gazebosim/ros_gz/tree/ros2/ros_gz_sim_demos

Also... hard to ask, but with best wishes, it seems `ros_gz`'s demo is now all in python. What do you think of working with python from this and now on? With power of chatgpt

Comment (woensug-choi) [reply dave_demos/README.md]: All other parts worked fine! Thank you so much for investigating the previous dave structure and attempt to follow how it was done 👍

Comment (rakeshv24) [reply dave_demos/README.md]: Sure thing, I'll make those changes!

**Comment by rakeshv24** [comment]
@woensug-choi, I've incorporated your comments into the commit.

**Review by woensug-choi — CHANGES_REQUESTED** [review]
So QUICK! Amazing..! 👍 

1. no need `_bare_model` for bare model. Will add `_something` as we add plugins or attachments (e.g. arms)

2. I've treid universal `upload_model.launch.py` and `model_in_empty_world.launch` at https://github.com/IOES-Lab/dave/tree/fuel-models-universal. 
```bash
ros2 launch dave_demos model_in_empty_world.py model_name:='nortek_dvl500_300'
ros2 launch dave_demos model_in_empty_world.py model_name:='mossy_cinder_block'
```
I think you can use `git merge fuel-models-universal` on your PR if things look ok (please test! and make mods if necessary). I haven't modified README for demos. you may commit on top of `fuel-models-universal` and merge to your PR.

3. Any idea why https://github.com/gazebosim/ros_gz/pull/492 didn't work? maybe it's not on release yet? added note on `package.xml` at `dave_model_description`

4. There was a typo error at README. `cd <path-to-dave_ws>/src/dave_model_description` -> `cd <path-to-dave_ws>/src/dave/dave_model_description`

**Comment by rakeshv24** [comment]
Awesome, thank you so for the quick and detailed review!

I like the idea of just using one universal launch file! I have merged the branch after making minor mods. 

Any idea why https://github.com/gazebosim/ros_gz/pull/492 didn't work? maybe it's not on release yet?
- It seems likely that the pull request isn’t included in the release yet. For now, let's use the hook and revisit this once it becomes available or if I find a way to make it work with Jazzy.

**Review by woensug-choi — APPROVED** [review]
Great! Our first PR! Thank you!!! 👍 
Take pleasure to push merge button

**Pull request merged**